cmds/core/fusermount/fusermount_linux.go

```diff
@@ -23,13 +27,19 @@ func dropPrivs() error {
 	return err
 }
 
-func restorePrivs() {
+func restorePrivs() error {
 	if unix.Getuid() == 0 {
-		return
+		return nil
 	}
 	// We're exiting, if there's an error, not much to do.
-	unix.Setfsuid(fileSystemUID)
-	unix.Setfsgid(fileSystemGID)
+	if err := unix.Setfsuid(fileSystemUID); err != nil {
```

Comment:
even if the setfsuid fails, the setfsgid might succeed. Consider using errors.Join